src/Microsoft.ML.Data/DataLoadSave/PartitionedFileLoader.cs

```diff
@@ -0,0 +1,726 @@
+                for (int col = 0; col < colLim; col++)
+                {
+                    var type1 = schema1.GetColumnType(col);
+                    var type2 = schema2.GetColumnType(col);
+                    if (!type1.Equals(type2))
+                    {
+                        return false;
+                    }
+                }
+
+                return true;
+            }
+
+            private Delegate MarshalGetter(Func<int, ValueGetter<int>> func, Type type, int col)
+            {
+                var returnType = typeof(ValueGetter<>).MakeGenericType(type);
+                var meth = func.Method;
+
+                var typedMeth = meth.GetGenericMethodDefinition().MakeGenericMethod(type);
+                return (Delegate)typedMeth.Invoke(func.Target, new object[] { col });
+            }
+
+            /// <summary>
+            /// Truncate path to the specified number of trailing directories.
+            /// </summary>
+            /// <param name="dirCount">Number of directories to retain.</param>
+            /// <param name="path">Path to truncate.</param>
+            /// <param name="truncPath">The resulting truncated path.</param>
+            /// <returns>true if the truncation was successful.</returns>
+            private bool TryTruncatePath(int dirCount, string path, out string truncPath)
+            {
+                truncPath = null;
+
+                // Remove directories that shouldn't be parsed.
+                var segments = Utils.SplitDirectories(path);
+                segments = segments.Skip(segments.Count() - dirCount - 1);
+
+                if (segments.Count() < dirCount - 1)
+                {
+                    Ch.Warning($"Path {path} did not have {dirCount} directories necessary for parsing.");
+                    return false;
+                }
+                
+                // Rejoin segments to create a valid path.
+                truncPath = String.Join(Path.DirectorySeparatorChar.ToString(), segments);
+                return true;
+            }
+
+
+            /// <summary>
+            /// Parse all column values from the directory path.
+            /// </summary>
+            /// <param name="path">The directory path to parse for name/value pairs.</param>
+            /// <param name="results">The resulting name value pairs.</param>
+            /// <returns>true if the parsing was successfull.</returns>
+            private bool TryParseValuesFromPath(string path, out List<string> results)
+            {
+                Contracts.CheckNonWhiteSpace(path, nameof(path));
+
+                results = null;
+
+                try
+                {
+                    results = _parent._pathParser.ParseValues(path).ToList();
+                    return true;
+                }
+                catch (FormatException e)
+                {
+                    Ch.Warning($"Could not parse column values from the path {path}.", e);
```

Comment:
>, e [](start = 85, length = 3)

Passing along an exception here is not meaningful, since this argument is intended to be used if the first argument had been a format string, but it's not, it's an already formatted string.

Reply:
If you really wanted to include information about the exception, it would be best to make them part of the format string.

---
In reply to: [187443722](https://github.com/dotnet/machinelearning/pull/61#discussion_r187443722) [](ancestors = 187443722)

Reply:
Thanks, I was blindly followed exception formatting.

Reply:
Resolved